internal/ansible/ansible_manager.go

```diff
@@ -0,0 +1,351 @@
+	dispatcherClient pb.DispatcherClient, configDir string) (*AnsibleManager, error) {
+	mappingRepository, err := mapping.NewMappingRepository(configDir)
+	if err != nil {
+		return nil, fmt.Errorf("ansible manager cannot initialize mapping repository: %w", err)
+	}
+
+	_, err = exec.LookPath("ansible")
+	if err != nil {
+		return nil, fmt.Errorf("flotta agent requires the ansible package to be installed")
+	}
+
+	return &AnsibleManager{
+		wg:                &sync.WaitGroup{},
+		dispatcherClient:  dispatcherClient,
+		ansibleDispatcher: dispatcher.NewAnsibleDispatcher(deviceID),
+		MappingRepository: mappingRepository,
+	}, nil
+}
+
+func MissingAttributeError(attribute string, metadata map[string]string) error {
+	return fmt.Errorf(missingAttributeMsg(attribute, metadata))
+}
+
+func missingAttributeMsg(attribute string, metadata map[string]string) string {
+	return fmt.Sprintf("missing attribute %s in message metadata %v", attribute, metadata)
+}
+
+// Set executor and stdoutcallback
+func setupPlaybookCmd(playbookCmd *playbook.AnsiblePlaybookCmd, buffOut, buffErr *bytes.Buffer) {
+	playbookExecutor := execute.NewDefaultExecute(
+		execute.WithWrite(io.Writer(buffOut)),
+		execute.WithWriteError(io.Writer(buffErr)),
+	)
+
+	playbookCmd.Exec = playbookExecutor
+	playbookCmd.StdoutCallback = "json"
+}
+
+func (a *AnsibleManager) HandlePlaybook(playbookCmd *playbook.AnsiblePlaybookCmd, d *pb.Data, timeout time.Duration) error {
+	var err error
+	buffOut := new(bytes.Buffer)
+	buffErr := new(bytes.Buffer)
+
+	setupPlaybookCmd(playbookCmd, buffOut, buffErr)
+
+	deviceConfigurationMessage := models.DeviceConfigurationMessage{}
+
+	if string(d.Content) == "" {
+		return fmt.Errorf("empty message. messageID: %s", d.MessageId)
+	}
+	err = json.Unmarshal(d.Content, &deviceConfigurationMessage)
+	if err != nil {
+		log.Error("Error while converting message content to map ", err)
+	}
+
+	// required fields
+	reqFields := &RequiredFields{}
+	payloadStr := deviceConfigurationMessage.AnsiblePlaybook
+	log.Infof("Handle Playbook Content message: %s", payloadStr)
+	found := false
+
+	responseTo := d.MessageId
+	if payloadStr == "" {
+		return fmt.Errorf("missing playbook string in message with messageID: %s", d.MessageId)
+	}
+	metadataMap := d.GetMetadata()
+	if reqFields.crcDispatcherCorrelationID, found = metadataMap[crcDispatcherAttribute]; !found {
+		return MissingAttributeError(crcDispatcherAttribute, metadataMap)
+	}
+
+	if reqFields.returnURL, found = metadataMap[returnURLAttribute]; !found {
+		return fmt.Errorf(missingAttributeMsg(returnURLAttribute, metadataMap))
+	}
+	playbookYamlFile := path.Join(dataDir, "ansible_playbook_"+d.MessageId+".yml")
+	err = os.WriteFile(playbookYamlFile, []byte(payloadStr), 0600)
+	if err != nil {
+		return fmt.Errorf("cannot create ansible playbook yaml file %s. Error: %v", playbookYamlFile, err)
+	}
+	defer os.Remove(playbookYamlFile)
+
+	/*
+	*******************************************
+	*  TODO : Verify playbook
```

Comment:
Due to the changes, I would create a task and track here the ID.

Reply:
I didn't add the Jira story ID because isn't accessible for non Red Hatters